tests/unittests/tests-sixlowpan_ctx/tests-sixlowpan_ctx.c

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright (C) 2014 Martine Lenders <mlenders@inf.fu-berlin.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ *
+ * @file
+ */
+#include <inttypes.h>
+#include <stdlib.h>
+
+#include "embUnit.h"
+
+#include "net/ng_ipv6/addr.h"
+#include "net/ng_sixlowpan/ctx.h"
+
+#include "unittests-constants.h"
+#include "tests-sixlowpan_ctx.h"
+
+#define DEFAULT_TEST_ID         (5)
+#define DEFAULT_TEST_PREFIX { { \
+            0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, \
+            0x08, 0x09, 0x0a, 0x0b, 0x0c, 0x0d, 0x0e, 0x0f \
+        } \
+    }
+#define DEFAULT_TEST_PREFIX_LEN (63)
+
+#define OTHER_TEST_ID         (12)
+#define OTHER_TEST_PREFIX   { { \
+            0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x06, \
+            0x18, 0x19, 0x1a, 0x1b, 0x1c, 0x1d, 0x1e, 0x1f \
+        } \
+    }
+#define WRONG_TEST_PREFIX   { { \
+            0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x05, \
+            0x18, 0x19, 0x1a, 0x1b, 0x1c, 0x1d, 0x1e, 0x1f \
+        } \
+    }
+
+static void tear_down(void)
+{
+    ng_sixlowpan_ctx_reset();
+}
+
+static void test_sixlowpan_ctx_update__wrong_id1(void)
+{
+    ng_ipv6_addr_t addr = DEFAULT_TEST_PREFIX;
+
+    TEST_ASSERT_NULL(ng_sixlowpan_ctx_update(NG_SIXLOWPAN_CTX_SIZE, &addr,
+                                             DEFAULT_TEST_PREFIX_LEN,
+                                             TEST_UINT16));
+    TEST_ASSERT_NULL(ng_sixlowpan_ctx_lookup_id(NG_SIXLOWPAN_CTX_SIZE));
+}
+
+static void test_sixlowpan_ctx_update__wrong_id2(void)
+{
+    ng_ipv6_addr_t addr = DEFAULT_TEST_PREFIX;
+
+    TEST_ASSERT_NULL(ng_sixlowpan_ctx_update(UINT8_MAX, &addr,
+                                             DEFAULT_TEST_PREFIX_LEN,
+                                             TEST_UINT16));
+    TEST_ASSERT_NULL(ng_sixlowpan_ctx_lookup_id(UINT8_MAX));
+}
+
+static void test_sixlowpan_ctx_update__wrong_prefix_len(void)
+{
+    ng_ipv6_addr_t addr = DEFAULT_TEST_PREFIX;
+
+    TEST_ASSERT_NULL(ng_sixlowpan_ctx_update(DEFAULT_TEST_ID, &addr,
+                                             0, TEST_UINT16));
```

Comment:
I know but as far as I can see the RFC does not explicitly forbid this. Anyhow, our implementation does so, so the test is valid as is.